### PR TITLE
removes dab

### PR DESCRIPTION
**Pull request by jcatjmeow**
### Intent of your Pull Request
It's been ages now that we voted to have our server go to MRP. I think it's time, finally, to actually start taking steps towards that goal.

We can try all we want to enforce rp via rules and administration but there will never ever be anything past LRP if we have shit like this in our code. And there's a lot of it, but dabbing was brought to my attention by Fluffe, so I'm starting with that.

There is zero reason in universe that

1. Dabbing would still be a thing people do
2. Dabbing would cause any spaceman to get spontaneous unexplained brain damage

It's completely LRP and immersion breaking and has no place on our server. It's basically OOC in IC which is already against the rules.

And please, before you comment, if your retort is "But [other thing] is in the game and that's even lowerrp" that does not excuse dabbing and I'll probably get around to removing that too, eventually.

If your retort is "improve don't remove" removal of bad features *is* improving the game more than keeping them around or shopping them into slightly less bad features.

If your retort is "I like LRP" see https://forums.yogstation.net/index.php?threads/increasing-our-roleplaying.20525/#post-174761. I'm sorry, but yogstation is not the place for you. There are plenty of other servers with minimal rp rules and many of these mechanics for you to enjoy.

If your retort is "I like dabbing" or "I don't think dabbing is lrp"... then I guess we have to agree to disagree 😩  

**Pr is both compiled and tested**

#### Changelog

:cl:  
rscdel: Dabbing has been removed.
/:cl:

Jamie Edit: Closes: #9765

### CODE REVIEW

**Comment by fluffe9911** [comment]
We had a server vote for this although it was 8 months ago so maybe opinions changed
image

**Comment by boodaliboo** [comment]
pr has been made like 3 times and closed all those times

**Comment by ynot01** [comment]
the brain damage was thought initially to be fine but it's become popular for greyshits to just dab themselves to death, which is pretty aids

alt could be just removing that aspect

**Comment by boodaliboo** [comment]
> the brain damage was thought initially to be fine but it's become popular for greyshits to just dab themselves to death, which is pretty aids
> 
> alt could be just removing that aspect

why does it matter if greyshirts do that this pr doesnt even stop them from doing so t-posing still exists

**Comment by ynot01** [comment]
just remembered the bit where people dab themselves into brain damage to get shit like voice of god and FUCK it's super annoying dealing with assistants with voice of god trauma

there are good uses of dabbing though like when you and the boys gamer a xeno or something and you dab on their body, good times

**Comment by boodaliboo** [comment]
> just remembered the bit where people dab themselves into brain damage to get shit like voice of god and FUCK it's super annoying dealing with assistants with voice of god trauma

then the proper solution is to either exclude voice of god trauma from self inflicted brain damage or just remove dabbing doing brain damage

**Comment by jcatjmeow** [comment]
I fully support removing the brain damage but I believe that is a half-measure to a full issue

**Comment by boodaliboo** [comment]
> I fully support removing the brain damage but I believe that is a half-measure to a full issue

i dont see how removing a action that people can also do in real life raises rp if its about people being annoying with it people will just find others way to piss people off

**Comment by jcatjmeow** [comment]
bood did you not read the pr? I detail exactly how.
Its not "an action people can do in real life" it's a meme, and a shitty meme. 
I don't mind people pissing each other off if it's in character and makes sense in universe. Dabbing is neither, it's OOC in IC

**Comment by jcatjmeow** [comment]
Also, yes, you can always do me dabs but like, that's a admin issue not a code issue

**Comment by Moja10** [comment]
Dab can be removed.

**Comment by maxion12345** [comment]
local coder angry at assistant for dabbing on him instead of playing the game correctly

**Comment by ThatLing** [comment]
based pr

**Review by ThatLing — APPROVED** [review]

**Comment by jcatjmeow** [comment]
> based pr

Based on what :3

> local coder angry at assistant for dabbing on him instead of playing the game correctly

Feel free to actually read the pr at your earliest convenience!

**Comment by CrazyDrakkon** [comment]
They read it and still disagree.

**Comment by boodaliboo** [comment]
> > based pr
> 
> Based on what :3
> 
> > local coder angry at assistant for dabbing on him instead of playing the game correctly
> 
> Feel free to actually read the pr at your earliest convenience!

you dont really need to see your reasoning to disagree with removing stuff

**Comment by jcatjmeow** [comment]
> you dont really need to see your reasoning to disagree with removing stuff

...what? Yes you fucking do

You really should understand someone else's position before disagreeing or agreeing with it. I was not aware this was a controversial opinion

**Comment by jcatjmeow** [comment]
> They read it and still disagree.

See bood and maxion. At least some of them didnt.

**Comment by boodaliboo** [comment]
> > you dont really need to see your reasoning to disagree with removing stuff
> 
> ...what? Yes you fucking do
> 
> You really should understand someone else's position before disagreeing or agreeing with it. I was not aware this was a controversial opinion

no you fucking dont, i can disagree with you removing something without knowing why. i dont care why your removing it i dont want it removed

**Comment by jcatjmeow** [comment]
> > > you dont really need to see your reasoning to disagree with removing stuff
> > 
> > 
> > ...what? Yes you fucking do
> > You really should understand someone else's position before disagreeing or agreeing with it. I was not aware this was a controversial opinion
> 
> no you fucking dont, i can disagree with you removing something without knowing why. i dont care why your removing it i dont want it removed

You *can* like how you *can* eat paste, but you shouldn't. It's not very intellectually honest to choose to not know why you don't agree with something.

**Comment by boodaliboo** [comment]
> > > > you dont really need to see your reasoning to disagree with removing stuff
> > > 
> > > 
> > > ...what? Yes you fucking do
> > > You really should understand someone else's position before disagreeing or agreeing with it. I was not aware this was a controversial opinion
> > 
> > 
> > no you fucking dont, i can disagree with you removing something without knowing why. i dont care why your removing it i dont want it removed
> 
> You _can_ like how _can_ eat paste, but it's not very intellectually honest.

im sorry i truthfully dont know what your trying to say here, are you just flinging insults at me now?

**Comment by jcatjmeow** [comment]
Capture
You would know what I was trying to say if you spent more than half a minute reading it

**Comment by boodaliboo** [comment]
> Capture
> You would know what I was trying to say if you spent more than half a minute reading it
oh you editted it i understand it now

**Comment by Smudged** [comment]
> > > you dont really need to see your reasoning to disagree with removing stuff
> > 
> > 
> > ...what? Yes you fucking do
> > You really should understand someone else's position before disagreeing or agreeing with it. I was not aware this was a controversial opinion
> 
> no you fucking dont, i can disagree with you removing something without knowing why. i dont care why your removing it i dont want it removed

Bood you gotta read the main body, if anything Jcat did list the reason why he wanted Dabbing removed 
"It's completely LRP and immersion breaking and has no place on our server. It's basically OOC in IC which is already against the rules."

I complety agree too, why would using a *dab emote or any emote give you brain damage, and why would that be conducive to how you play the game other than "it's a meme"

**Comment by Takahiru** [comment]
Bood please state your reason for why you dont want it removed otherwise I see no reason why your point should be taken into consideration?

**Comment by boodaliboo** [comment]
> Bood please state your reason for why you dont want it removed otherwise I see no reason why your point should be taken into consideration?

cool

**Comment by maxion12345** [comment]
I'm gonna be honest here, removing this shit isn't going to help the server transition towards mrp. I guarantee that nobody here actually fucking thinks "bro spaceman dab means we're a lrp server" like that. And while the game is set in the 26th century, I sure as fuck am from the 21st century and therefore want my relateable comedy.

I mean, we can't go around removing shit because "it's a meme", otherwise we'll be deleting half our codebase and banning oak. Focus on the game mechanics that are at the core of our server if you want to change things up. 

Besides, if this is merged I'm 200 percent making a ahk script to *me dab and *spin at the same time.

It'll be even more autistic because then there's no brain damage to stop me, ehehehhehe

**Comment by jcatjmeow** [comment]
>I mean, we can't go around removing shit because "it's a meme"

We can, I'm doing it right now. Of course, it's because "memes from today break ooc in ic rules, and are lrp" but go off

>otherwise we'll be deleting half our codebase and banning oak.

I see nothing wrong with this, a lot of our codebase is inherited from tg which is lrp and could do with adjustment. And if oak was banned from the server he'd contribute exactly as much as he is right now. :p But yeah, our codebase is so bloated and people are always so allergic to deletion that it keeps getting worse and worse, I think some trimming of the lrp stuff is in order.

>removing this shit isn't going to help the server transition towards mrp.

Disagree, because

>while the game is set in the 26th century, I sure as fuck am from the 21st century and therefore want my relateable comedy.

This is a lowrp mindset, basically. By encouraging this behaviour you encourage people to not be in character or roleplay or anything. So removing this shit is absolutely a step in the right direction.

Also dabbing isn't comedy lmao

**Comment by maxion12345** [comment]
> > I mean, we can't go around removing shit because "it's a meme"
> 
> We can, I'm doing it right now. Of course, it's because "memes from today break ooc in ic rules, and are lrp" but go off
> 
> > otherwise we'll be deleting half our codebase and banning oak.
> 
> I see nothing wrong with this, a lot of our codebase is inherited from tg which is lrp and could do with adjustment. And if oak was banned from the server he'd contribute exactly as much as he is right now. :p But yeah, our codebase is so bloated and people are always so allergic to deletion that it keeps getting worse and worse, I think some trimming of the lrp stuff is in order.
> 
> > removing this shit isn't going to help the server transition towards mrp.
> 
> Disagree, because
> 
> > while the game is set in the 26th century, I sure as fuck am from the 21st century and therefore want my relateable comedy.
> 
> This is a lowrp mindset, basically. By encouraging this behaviour you encourage people to not be in character or roleplay or anything. So removing this shit is absolutely a step in the right direction.
> 
> Also dabbing isn't comedy lmao

Alright. Remove holoparas. They are memes based off of a damn anime. Literally one big fucking reference.
While your at it, better remove the jukebox and disco ball. All the songs there are just memes, like sandstorm.
You want to remove another meme? Removing standing on tables with an esword giving you dummy high block. 
What about the circle hand? Also, better remove slapping. Having a way to attack players without doing damage would only encourage LRP, obviously. While we're at it, delete the medbay vendors and make sleepers only useable by doctors. Same with cloning and cyro, so nobody can just bypass the roleplay anymore. 

My point here is that if you want to change the game, focus on the shit that matters. You wont be changing anything with this PR. And while you might be doing it to make a point, the only point I see is that we should remove anything "meme" and force our players to conform to a character within the game. I sure as fuck dont want that. If you remove ways people can have fun or have a good laugh, people aren't going to be playing this server anymore. 

I also think we cant have MRP without a bit of LRP. That's the appeal of MRP anyways, your not forced to roleplay standards and still can goof off and do dumb shit if you want, so long as it doesn't really affect the round in any big way. Otherwise, we'll end up banning assistants for making little maint homes because "Thats powergaming" and "Assistants should assist". Did an assistant just build a skateboard and tables and is grinding down the hallways? Not conforming to roleplay, remove the skateboards. And the tables while we're at it, so players must hand items to eachother (To encourage roleplay, obviously).
TL;DR
>Removing shit for being "meme" is retarded
>Forcing players to conform to a character by removing fun shit will simply backfire and is also retarded
>You cant have MRP without some LRP

**Comment by jcatjmeow** [comment]
> Alright. Remove holoparas. They are memes based off of a damn anime. Literally one big fucking reference.

Agree 90%. But this is a pr about dabbing.
> While your at it, better remove the jukebox and disco ball. All the songs there are just memes, like sandstorm.

Agree 100%. But this is a pr about dabbing.
> You want to remove another meme? Removing standing on tables with an esword giving you dummy high block.

Agree 100%. But this is a pr about dabbing.
> What about the circle hand? 

Agree 50%. But this is a pr about dabbing.
Also, for circle hand specifically, I was talking with people in discord about removing it already.
Anyway, this is all "whataboutism", a common logical fallacy. You're saying whatabout this whatabout that, sure I will get around to removing those things if I feel like they are LRP enough to cross the line, which will take some discussion with regulars and people who are good roleplayers and stuff. But like, all those things are irrelevant for the discussion in this one, atomized pr, which is about specifically dabbing. 

I'm only one guy, and each pr should only really have one change in it, and I have to think very carefully about what I choose to remove or change before I do it (otherwise I'd be banging these out like hotcakes). Dabbing took less time because I very quickly came to the conclusion that it has absolutely no merit. Things like, for example the circle hand (hey that's the thing you are talking about) are obviously contemporary memes, but you can also use the emote to just make your hand in a circle for whatever reason, idk, so it's not as black and white (I am at least 50% on board for circle hand)

>Also, better remove slapping. Having a way to attack players without doing damage would only encourage LRP, obviously. 

Quite the opposite: Being able to roleplay out attacking someone or being physical with them without causing any ill effects is an excellent use of such an emote.

>While we're at it, delete the medbay vendors and make sleepers only useable by doctors. Same with cloning and cyro, so nobody can just bypass the roleplay anymore.

Medical supplies exist, and I think that very good medicine is in the realms of 'suspension of disbelief' and they are standard in a sci fi setting, along with cloning.

> My point here is that if you want to change the game, focus on the shit that matters. 

This does matter

>You wont be changing anything with this PR. 

I'll be removing the ability for people to dab without using *me

>And while you might be doing it to make a point

I'm not

>the only point I see is that we should remove anything "meme" and force our players to conform to a character within the game. I sure as fuck dont want that. If you remove ways people can have fun or have a good laugh, people aren't going to be playing this server anymore.

We can keep goofy, wacky, silly things in the game, but some things are only there to foster a LRP environment and have no value elswewise. We don't have to throw the baby out with the bathwater

> I also think we cant have MRP without a bit of LRP. 

WHAT
ok, I think I get what you are saying? It's not like we can weight the server as a whole as MRP or LRP, individual things are one or the other. And some things will always be more 'meta' than others, I get that. I agree. But we can totally have a medium amount of roleplay without having shit like dabbing and people saying lol and 420. All three of these things are against ooc in ic rules on paper, but dabbing is still *in* the code for god knows why

>That's the appeal of MRP anyways, your not forced to roleplay standards

Well uh... I guess this is the problem with having nebulous terms like MRP and LRP and HRP. Not everyone will have the same definition. 

For me, MRP means that people check their ooc shit at the door. You don't have to write 1,000 lines of description text about your furry wolf cock but you have to actually commit to being in the headspace and not just run around being Jcatjmeow the video gamer doing whatever you want like it's a singleplayer game.

>and still can goof off and do dumb shit if you want, so long as it doesn't really affect the round in any big way.

I like goofing off and 'dumb shit' but not LRP stuff like dabbing. 

>Otherwise, we'll end up banning assistants for making little maint homes because "Thats powergaming" 

Depending on what you mean by 'maint homes' this is something we already do

>and "Assistants should assist". 

I mean they kinda should yeah. No greytiding is already a rule

>Did an assistant just build a skateboard and tables and is grinding down the hallways? Not conforming to roleplay, remove the skateboards.

Why isn't that conforming to roleplay? That meets the bar for me. They aren't saying HAHA BIG CHUNGUS REDDIT GOLD DAB and talking to me about HEY ADMINS LOOK AT MY COCK ya know? They are playing a character who likes to skateboard. Go off! 
Sounds like a fun rp gimmick to me! If all assistants were like that I think the server would be better off!

>And the tables while we're at it, so players must hand items to eachother (To encourage roleplay, obviously).

You are talking to the guy who did attempt to push a "remove all tables pr" 😩 (it was a joke)
But I do think there is no problem with that, I don't see any issue with players handing items to each other. It's something that can already be done, you can offer people items and they get a prompt to take it if they want. I love that stuff! 


Sorry for the long response but I felt like I had things to say about the stuff and things you said

**Comment by boodaliboo** [comment]
> > Alright. Remove holoparas. They are memes based off of a damn anime. Literally one big fucking reference.
> 
> Agree 90%. But this is a pr about dabbing.
> 
> > While your at it, better remove the jukebox and disco ball. All the songs there are just memes, like sandstorm.
> 
> Agree 100%. But this is a pr about dabbing.
> 
> > You want to remove another meme? Removing standing on tables with an esword giving you dummy high block.
> 
> Agree 100%. But this is a pr about dabbing.
> 
> > What about the circle hand?
> 
> Agree 50%. But this is a pr about dabbing.
> Also, for circle hand specifically, I was talking with people in discord about removing it already.
> Anyway, this is all "whataboutism", a common logical fallacy. You're saying whatabout this whatabout that, sure I will get around to removing those things if I feel like they are LRP enough to cross the line, which will take some discussion with regulars and people who are good roleplayers and stuff. But like, all those things are irrelevant for the discussion in this one, atomized pr, which is about specifically dabbing.
> 
> I'm only one guy, and each pr should only really have one change in it, and I have to think very carefully about what I choose to remove or change before I do it (otherwise I'd be banging these out like hotcakes). Dabbing took less time because I very quickly came to the conclusion that it has absolutely no merit. Things like, for example the circle hand (hey that's the thing you are talking about) are obviously contemporary memes, but you can also use the emote to just make your hand in a circle for whatever reason, idk, so it's not as black and white (I am at least 50% on board for circle hand)
> 
> > Also, better remove slapping. Having a way to attack players without doing damage would only encourage LRP, obviously.
> 
> Quite the opposite: Being able to roleplay out attacking someone or being physical with them without causing any ill effects is an excellent use of such an emote.
> 
> > While we're at it, delete the medbay vendors and make sleepers only useable by doctors. Same with cloning and cyro, so nobody can just bypass the roleplay anymore.
> 
> Medical supplies exist, and I think that very good medicine is in the realms of 'suspension of disbelief' and they are standard in a sci fi setting, along with cloning.
> 
> > My point here is that if you want to change the game, focus on the shit that matters.
> 
> This does matter
> 
> > You wont be changing anything with this PR.
> 
> I'll be removing the ability for people to dab without using *me
> 
> > And while you might be doing it to make a point
> 
> I'm not
> 
> > the only point I see is that we should remove anything "meme" and force our players to conform to a character within the game. I sure as fuck dont want that. If you remove ways people can have fun or have a good laugh, people aren't going to be playing this server anymore.
> 
> We can keep goofy, wacky, silly things in the game, but some things are only there to foster a LRP environment and have no value elswewise. We don't have to throw the baby out with the bathwater
> 
> > I also think we cant have MRP without a bit of LRP.
> 
> WHAT
> ok, I think I get what you are saying? It's not like we can weight the server as a whole as MRP or LRP, individual things are one or the other. And some things will always be more 'meta' than others, I get that. I agree. But we can totally have a medium amount of roleplay without having shit like dabbing and people saying lol and 420. All three of these things are against ooc in ic rules on paper, but dabbing is still _in_ the code for god knows why
> 
> > That's the appeal of MRP anyways, your not forced to roleplay standards
> 
> Well uh... I guess this is the problem with having nebulous terms like MRP and LRP and HRP. Not everyone will have the same definition.
> 
> For me, MRP means that people check their ooc shit at the door. You don't have to write 1,000 lines of description text about your furry wolf cock but you have to actually commit to being in the headspace and not just run around being Jcatjmeow the video gamer doing whatever you want like it's a singleplayer game.
> 
> > and still can goof off and do dumb shit if you want, so long as it doesn't really affect the round in any big way.
> 
> I like goofing off and 'dumb shit' but not LRP stuff like dabbing.
> 
> > Otherwise, we'll end up banning assistants for making little maint homes because "Thats powergaming"
> 
> Depending on what you mean by 'maint homes' this is something we already do
> 
> > and "Assistants should assist".
> 
> I mean they kinda should yeah. No greytiding is already a rule
> 
> > Did an assistant just build a skateboard and tables and is grinding down the hallways? Not conforming to roleplay, remove the skateboards.
> 
> Why isn't that conforming to roleplay? That meets the bar for me. They aren't saying HAHA BIG CHUNGUS REDDIT GOLD DAB and talking to me about HEY ADMINS LOOK AT MY COCK ya know? They are playing a character who likes to skateboard. Go off!
> Sounds like a fun rp gimmick to me! If all assistants were like that I think the server would be better off!
> 
> > And the tables while we're at it, so players must hand items to eachother (To encourage roleplay, obviously).
> 
> You are talking to the guy who did attempt to push a "remove all tables pr" 😩 (it was a joke)
> But I do think there is no problem with that, I don't see any issue with players handing items to each other. It's something that can already be done, you can offer people items and they get a prompt to take it if they want. I love that stuff!
> 
> Sorry for the long response but I felt like I had things to say about the stuff and things you said

you should shorten these

**Comment by Takahiru** [comment]
About the holoparas thing--yes they are based off of something else but they were made into something cool and can stand by itself as a sort of syndicate item, should we remove all items based on another thing the game wouldnt exist, even the 'cryptographic sequencer' name comes from a batman game. Dabbing is a pointless feature which dosen't really add much. Goddamn the dab wars are heated though.

**Comment by JamieD1** [comment]
Council Vote time

File: code/__HELPERS/matrices.dm [hunk 1]
```diff
@@ -27,33 +27,6 @@
...
-	DAB_COMMENCE.Turn(angle) // dab angle to matrix
-
-	// Dab animation 
-	animate(src, transform = DAB_COMMENCE, time = speed, loops ) // dab to hold angle
-	if(hold_seconds > 0)
-		sleep(hold_seconds) // time to hold the dab before going back
-	if(!stay) // if stay param is true dab doesn't return
-		animate(src, transform = DAB_RETURN, time = speed * 1.5, loops ) // reverse dab to starting position , slower
-
 //Dumps the matrix data in format a-f
```
Comment (JamieD1) [inline code/__HELPERS/matrices.dm:56]:
```suggestion
/atom/proc/DabAnimation(speed = 1, loops = 1, direction = 1 , hold_seconds = 0  , angle = 1 , stay = FALSE) // Hopek 2019  
	// By making this in atom/proc everything in the game can potentially dab. You have been warned.
	if(hold_seconds > 9999) // if you need to hold a dab for more than 2 hours intentionally let me know.
		return
	if(hold_seconds > 0)
		hold_seconds = hold_seconds * 10 // Converts seconds to deciseconds 
	if(angle == 1) //if angle is 1: random angle. Else take angle
		angle = rand(25,50)
	if(direction == 1) // direciton:: 1 for random pick, 2 for clockwise , 3 for anti-clockwise
		direction = pick(2,3)
	if(direction == 3) // if 3 then counter clockwise
		angle = angle * -1
	if(speed == 1) // if speed is 1 choose random speed from list
		speed = rand(3,5)

	// dab matrix here
	var/matrix/DAB_COMMENCE = matrix(transform)
	var/matrix/DAB_RETURN = matrix(transform)
	DAB_COMMENCE.Turn(angle) // dab angle to matrix

	// Dab animation 
	animate(src, transform = DAB_COMMENCE, time = speed, loops ) // dab to hold angle
	if(hold_seconds > 0)
		sleep(hold_seconds) // time to hold the dab before going back
	if(!stay) // if stay param is true dab doesn't return
		animate(src, transform = DAB_RETURN, time = speed * 1.5, loops ) // reverse dab to starting position , slower
//Dumps the matrix data in format a-f
```

Comment (JamieD1) [reply code/__HELPERS/matrices.dm:56]: Otherwise gun animations will break

Comment (ThatLing) [reply code/__HELPERS/matrices.dm:56]: Why are gun animations dependant on the dab animation?

**Comment by JamieD1** [comment]
Goodbye Dab.
